src/components/menu-bar/account-nav.jsx

```diff
@@ -0,0 +1,125 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import {connect} from 'react-redux';
+
+import Dropdown from '../dropdown/dropdown.jsx';
+import dropdownStyles from '../dropdown/dropdown.css';
+
+import styles from './account-nav.css';
+
+const AccountNav = ({
+    accountNavOpen,
+    classroomId,
+    isEducator,
+    isStudent,
+    onOpenAccountNav,
+    onCloseAccountNav,
+    onLogOut,
+    profileUrl,
+    // thumbnailUrl, // not currently needed
+    username
+}) => (
+    <div className="account-nav">
+        <a
+            className={classNames(
+                styles.userInfo,
+                {[styles.open]: accountNavOpen}
+            )}
+            href="#"
+            onClick={onOpenAccountNav}
+        >
+            {/* thumbnail image currently disabled
+                <img
+                    className={styles.avatar}
+                    src={thumbnailUrl}
+                />
+            */}
+            <span className="profile-name">
+                {username}
+            </span>
+        </a>
+        <Dropdown
+            as="ul"
+            className={process.env.SCRATCH_ENV}
+            isOpen={accountNavOpen}
+            onRequestClose={onCloseAccountNav}
+        >
+            <li>
+                <a href={profileUrl}>
+                    <FormattedMessage id="general.profile" />
+                </a>
+            </li>
+            <li>
+                <a href="/mystuff/">
+                    <FormattedMessage id="general.myStuff" />
+                </a>
+            </li>
+            {isEducator ? [
+                <li key="my-classes-li">
+                    <a href="/educators/classes/">
+                        <FormattedMessage id="general.myClasses" />
+                    </a>
+                </li>
+            ] : []}
+            {isStudent ? [
+                <li key="my-class-li">
+                    <a href={`/classes/${classroomId}/`}>
+                        <FormattedMessage id="general.myClass" />
+                    </a>
+                </li>
+            ] : []}
+            <li>
+                <a href="/accounts/settings/">
+                    <FormattedMessage id="general.accountSettings" />
+                </a>
+            </li>
+            <li className={dropdownStyles.divider}>
+                <a
+                    href="#"
+                    onClick={onLogOut}
+                >
+                    <FormattedMessage id="navigation.signOut" />
+                </a>
+            </li>
+        </Dropdown>
+    </div>
+);
+
+AccountNav.propTypes = {
+    accountNavOpen: PropTypes.bool,
+    classroomId: PropTypes.string,
+    isEducator: PropTypes.bool,
+    isStudent: PropTypes.bool,
+    onCloseAccountNav: PropTypes.func,
+    onLogOut: PropTypes.func,
+    onOpenAccountNav: PropTypes.func,
+    profileUrl: PropTypes.string,
+    // thumbnailUrl: PropTypes.string, // not currently needed
```

Comment:
Let's remove if it's unnecessary

Reply:
So I wanted to ask about this. Do we want to show the user thumbnail here? There's one in the beta dummy bar, which made me imagine that was part of the design intention. (I actually don't have access to the GUI project in invision, just asked Carl for it.)

Reply:
Yes we do want to display it.